src/components/shared/Search/Search.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react';
+import { filter, range, reduce } from 'lodash';
+import { Search, Grid } from 'semantic-ui-react';
+import { searchArticle } from '../../../actions/search.action';
+import './search.scss';
+
+class SearchComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoading: false,
+      results: [],
+      value: '',
+      getResults: {
+        articles: {},
+        authors: {},
+        tags: {},
+      },
+    };
+  }
+
+  handleResultSelect = (e, { result }) =>
+    this.setState({ value: result.title });
+
+  filterResultByName = arr => {
+    return reduce(
+      arr,
+      (acc, cur, name) => {
+        const results = filter(cur.results);
+        if (results.length) acc[name] = { name, results };
+        return acc;
+      },
+      {}
+    );
+  };
+
+  updateCategoriesWithSearchResult = result => {
+    return range(0, 3).reduce((acc, cur) => {
+      const name = ['articles', 'authors', 'tags'];
+      const results = [...result];
+      acc[name[cur]] = {
+        name: name[cur],
+        results: results[cur],
+      };
+      return acc;
+    }, {});
+  };
+
+  handleSearchChange = async (e, { value: currentVal }) => {
+    this.setState({ isLoading: true, value: currentVal });
+    const { authorFilter, articleFilter, keywordFilter } = await searchArticle(
+      currentVal
+    );
+    const newAuthors = [];
+    const newArticles = [];
+    const newTags = [];
+    keywordFilter.forEach(tags => {
+      newTags.push({
+        title: tags.article.title,
+        description: tags.article.body.substring(0, 55),
+        image: tags.article.author.image_url,
+        price: tags.keyword,
+      });
+    });
+    authorFilter.forEach(author => {
+      newAuthors.push({
+        title: author.first_name + author.last_name,
+        description: author.bio,
+        image: author.image_url,
+      });
+    });
+    articleFilter.forEach(article => {
+      newArticles.push({
+        title: article.title,
+        description: article.body.substring(0, 55),
+        image: article.author.image_url,
+      });
+    });
+    this.setState({
+      getResults: { articles: newArticles, authors: newAuthors, tags: newTags },
+    });
+    const { getResults } = this.state;
+
+    return setTimeout(() => {
+      const { articles, authors, tags } = getResults;
+      const source = this.updateCategoriesWithSearchResult([
+        articles,
+        authors,
+        tags,
+      ]);
+      const filteredResults = this.filterResultByName(source);
+
+      return this.setState({
+        isLoading: false,
+        results: filteredResults,
+      });
+    }, 500);
```

Comment:
Are we waiting 500 milliseconds now? @Akinmyde

Reply:
Yes, I found out that 3000 milliseconds will be a lot